model/scratch/dffml_model_scratch/model/slr.py

```diff
@@ -0,0 +1,153 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2019 Intel Corporation
+'''
+Description of what this model does
+'''
+import os
+import abc
+import hashlib
+import json
+from typing import AsyncIterator, Tuple, Any, List, Optional, NamedTuple, Dict
+from statistics import mean
+import numpy as np
+
+from dffml.repo import Repo
+from dffml.source.source import Sources
+from dffml.feature import Features
+from dffml.accuracy import Accuracy
+from dffml.model.model import ModelConfig, ModelContext, Model
+from dffml.util.entrypoint import entry_point
+from dffml.util.cli.arg import Arg
+
+
+class SLRConfig(ModelConfig, NamedTuple):
+    predict: str
+    directory: str
+
+
+class SLRContext(ModelContext):
+    '''
+    Model wraping scratch API
+    '''
+    def __init__(self, parent):
+        super().__init__(parent)
+        self.xData = np.array([])
+        self.yData = np.array([])
+        self.key = None
+        self.regression_line = self.get_regression_line()
```

Comment:
You should remove `self.key` and `self.regression_line`, the methods `get_regression_line` and `set_regression_line` are [setters and getters](https://guide.freecodecamp.org/java/getters-and-setters/) which is a concept allowing us to remove usages of variables and replace them with methods which give us greater control.

Reply:
can't we use [property](https://docs.python.org/2/library/functions.html#property) like:

```python
@property
def regression_line(self):
    pass

# if there is a need for a setter 
@regression_line.setter
def regression_line(self, val)
    pass
instead of java like setter and getter to be more pythonic?

Reply:
I was hoping we could, but since this is dependent on `features` we can't with the current implementation. Which is now leading me to think. If the `Model` context is always dependent on the `features` maybe we should remove `features` from the arguments to `train`, `accuracy`, and `predict` and we'd end up with a cleaner API.

I think that this is the direction we will go: #101